packages/react/select/src/Select.tsx

```diff
@@ -297,21 +297,25 @@ interface SelectContentProps extends SelectContentImplProps {}
 const SelectContent = React.forwardRef<SelectContentElement, SelectContentProps>(
   (props: ScopedProps<SelectContentProps>, forwardedRef) => {
     const context = useSelectContext(CONTENT_NAME, props.__scopeSelect);
-    const fragmentRef = React.useRef(new DocumentFragment());
+    const [fragment, setFragment] = React.useState<DocumentFragment>();
+
+    useLayoutEffect(() => {
+      setFragment(new DocumentFragment());
+    }, []);
 
     return context.open ? (
       <SelectContentImpl {...props} ref={forwardedRef} />
-    ) : (
+    ) : fragment ? (
       ReactDOM.createPortal(
-        <Collection.Slot scope={props.__scopeSelect}>
-          {/* @ts-ignore: This is to avoid the "SelectViewport must be inside "SelectContent" error */}
-          <SelectContentContextProvider scope={props.__scopeSelect}>
+        // @ts-ignore: This is to avoid the "SelectViewport must be inside "SelectContent" error
+        <SelectContentContextProvider scope={props.__scopeSelect}>
+          <Collection.Slot scope={props.__scopeSelect}>
```

Comment:
i had to reorder these otherwise the collection slot was trying to slot onto the provider

Reply:
Nicely caught!